opm/simulators/utils/MPIPacker.hpp

```diff
@@ -26,19 +26,27 @@
 
 #include <bitset>
 #include <cstddef>
+#include <limits>
 #include <string>
 
+
 namespace Opm {
 namespace Mpi {
 namespace detail {
 
+static std::size_t mpi_buffer_size(const std::size_t bufsize, const std::size_t position) {
+    assert (bufsize >= position);
+    return static_cast<int>(std::min(bufsize-position,
+                                     static_cast<std::size_t>(std::numeric_limits<int>::max())));
```

Comment:
I think we should rather throw an exception if `bufsize-position > static_cast<std::size_t>(std::numeric_limits<int>::max())` holds.

Reply:
Actually I believe we need to allow bufsize-position > intmax, we limit it to the latter only to avoid problems in MPI_Pack, but the buffer is actually larger. Of course, trying to pack more than intmax bytes in one go will fail, but this should be picked up in the packSize call (below). 

Will change the assert to throw though.